en/guide/extending.md

```diff
@@ -15,39 +15,35 @@ Video Streaming | Developers can implement new types of video streaming (e.g. HL
 
 ## Support Custom Camera Device
 
-In order to support a new type of camera a custom class must be derived from `CameraDevice`. In addition, code must be added to detect the new device type and to instantiate the new custom camera device when this happens. The sections below detail these steps.
+Developers may want to add support for a new type of camera device that is not currently supported by DCM. To support a custom type of camera, a `PluginXXX` class derived from `PluginBase` and `CameraDeviceXXX` class derived from `CameraDevice` must be implemented. An example of a plugin for a custom camera device with classes `PluginCustom`(https://github.com/Dronecode/camera-manger/tree/master/plugins) and `CameraDeviceCustom`(https://github.com/Dronecode/camera-manger/tree/master/plugins) is added in the project and may be referenced.
+The sections below detail these steps.
 
-### 1. Extend CameraDevice Class
+### 1. Extend PluginBase Class
 
-To add support for new type of camera device in DCM, a custom class must be derived from `CameraDevice`. 
-An example `CameraDeviceCustom` (in green) can be seen in the [class diagram](../guide/architecture.md#class_diagram). 
-The class `CameraDeviceCustom` represents a custom type of camera device.
+Plugins are self-registering objects that discovers, enlists and creates camera devices.
+
+**Action**: `PluginCustom` must implement the pure virtual functions of the base class `PluginBase`.
```

Comment:
I'm not sure we need to have these "Action" headings. It might also be worth having the functions that must be implemented listed here.